src/core/lib/iomgr/tcp_server_posix.cc

```diff
@@ -205,8 +238,7 @@ static grpc_error_handle CreateEventEngineListener(
           GPR_ASSERT(gpr_atm_no_barrier_load(&s->refs.count) == 0);
           grpc_event_engine::experimental::RunEventEngineClosure(
               shutdown_complete, absl_status_to_grpc_error(status));
-          delete s->fd_handler;
-          delete s;
+          finish_shutdown(s);
```

Comment:
Is finish_shutdown just equivalent to `delete s->fd_handler; delete s;`  for the EventEngine case ?

Reply:
Not quite. In my PR description, I outlined that one of the key changes here is that none of the constituent objects are destroyed until the server is destroyed. That includes the mutex, the listeners, etc.. Those are now destroyed in `finish_shutdown`.